### PR TITLE
Add expected inputs and outputs for datastore extension

### PR DESCRIPTION
**Pull request by tkyi**
Resolves https://github.com/screwdriver-cd/screwdriver-datastore-base/issues/3

Adding expected parameters and results for get, save, update, and scan methods

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,6 +33,64 @@ store.get({ params: { id: 1 } }, (err, data) => {
...
+## Expected inputs and outputs for datastore implementations
+
+### get
+
+Obtain a single record given an id. If the record does not exist, it will return `null`.
+
+**Arguments**
+
+* `config` - An `object`. Each of its properties defines your get operation
+* `config.table` - A `string`. The DynamoDB table name
```
Comment (FenrirUnbound) [inline README.md]: Replace `DynamoDB`

File: README.md [hunk 2]
```diff
@@ -33,6 +33,64 @@ store.get({ params: { id: 1 } }, (err, data) => {
...
+###  update
+
+Update a record in the datastore. Returns `null` if the record does not exist.
+
+**Arguments**
+
+* `config` - An `object`. Each of its properties defines your save operation
+* `config.table` - A `string`. The datastore table name
+* `config.params` - An object. Each of its properties defines the save parameters
+* `config.params.id` - A `number`. The ID to associate the data with
```
Comment (FenrirUnbound) [inline README.md]: IIRC, it's a string

File: README.md [hunk 3]
```diff
@@ -33,6 +33,64 @@ store.get({ params: { id: 1 } }, (err, data) => {
...
+* `config.params.data` - An object. This is what will be saved in the datastore
+* `callback(err, result)` - A callback which is called when the task is completed. It receives the `err` and `result`, where `result` is the data that was saved in the datastore.
```
Comment (FenrirUnbound) [inline README.md]: Would it be too redundant if we also state the "Returns `null` if the record does not exist." message here?

Comment (tkyi) [inline README.md]: Sounds reasonable

File: README.md (hunk 2 above)
Comment (tkyi) [inline README.md]: Ooh okay we need to fix this in imdb too then

**Comment by FenrirUnbound** [comment]
👍 

Looks good to me

4bjozhu

File: README.md [hunk 4]
```diff
@@ -33,6 +33,64 @@ store.get({ params: { id: 1 } }, (err, data) => {
     // do something....
 });
 ```
+## Expected inputs and outputs for datastore implementations
+
+### get
+
+Obtain a single record given an id. Returns `null` if the record does not exist.
```
Comment (d2lam) [inline README.md]: What if the table does not exist?

File: README.md [hunk 5]
```diff
@@ -33,6 +33,64 @@ store.get({ params: { id: 1 } }, (err, data) => {
...
+
+**Arguments**
+
+* `config` - An `object`. Each of its properties defines your get operation
+* `config.table` - A `string`. The datastore table name
+* `config.params` - An object. Each of its properties defines the get parameters
```
Comment (d2lam) [inline README.md]: An `object`

File: README.md [hunk 6]
```diff
@@ -33,6 +33,64 @@ store.get({ params: { id: 1 } }, (err, data) => {
...
+
+###  update
+
+Update a record in the datastore. Returns `null` if the record does not exist.
+
+**Arguments**
+
+* `config` - An `object`. Each of its properties defines your save operation
+* `config.table` - A `string`. The datastore table name
+* `config.params` - An object. Each of its properties defines the save parameters
```
Comment (d2lam) [inline README.md]: An `object`

**Comment by d2lam** [comment]
👍

**Pull request merged**